src/users/userSelectors.js

```diff
@@ -41,6 +41,13 @@ export const getAllUsersByEmail = createSelector(getAllUsers, allUsers =>
   }, {}),
 );
 
+export const getUsersByEmail: Selector<UserEmailMap> = createSelector(getUsers, (users = []) =>
+  users.reduce((usersById, user) => {
+    usersById[user.email] = user;
+    return usersById;
+  }, ({ ...undefined }: UserEmailMap)),
```

Comment:
This fixes: "Cannot cast object literal to UserEmailMap because inexact object literal [1] is incompatible with exact  UserEmailMap [2]."

Because `{}` is considered 'unsealed object'
https://flow.org/en/docs/types/objects/#toc-unsealed-objects

A concept in Flow that might be removed as not very intuitive.

Reply:
Huh fascinating!

Ah, I see you've made `UserEmailMap` exact. I've shied away from making many types with "indexer properties" exact, because I keep getting strange errors from them. This may be a very useful workaround for that problem!

Let's be sure to mention that either in a comment and/or the commit message. Also you mentioned in audio chat that you found this workaround in an issue thread on Flow. That link will be super useful to have written down.

Reply:
bump:

> Also you mentioned in audio chat that you found this workaround in an issue thread on Flow. That link will be super useful to have written down.

Reply:
The exact issue I found was:
https://github.com/facebook/flow/issues/7343
This does not have any extra information that is worth linking to, besides the fact it named the problem (unsealed objects) which made me read the docs.

Reply:
There are other GitHub issues that discuss sealed/unsealed objects, like:
https://github.com/facebook/flow/issues/2977

I don't think linking them directly is of any specific benefit since the main benefit is identifying the cause and knowing what term to search for.